dumper-integration-tests/redshift-tests/src/main/java/com/google/edwmigration/dumper/base/TestBase.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.edwmigration.dumper.base;
+
+import static java.lang.String.format;
+import static java.lang.System.lineSeparator;
+
+import com.google.common.base.Joiner;
+import com.google.common.collect.LinkedHashMultiset;
+import com.opencsv.CSVParser;
+import com.opencsv.CSVParserBuilder;
+import org.junit.Assert;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Base class with general values for all TestNG test suites */
+public abstract class TestBase {
+
+  public static final CSVParser CSV_PARSER = new CSVParserBuilder().withEscapeChar('\0').build();
+  private static final Logger LOGGER = LoggerFactory.getLogger(TestBase.class);
+
+  /**
+   * @param dbMultiset List of extracted from DB items
+   * @param csvMultiset List of uploaded from Avro items
+   */
+  public static void assertMultisetsEqual(
```

Comment:
Can we use Maps.difference() on a Map<T, Integer>?

Reply:
**Conclusion**
I've tried this approach with `Maps.difference()` and it doesn't give me a desirable result.
Because it's keeping custom objects as Keys and instance counters as Values, this algorithm misses pretty obvious test cases, thus the test runs and indicates false-positive result.

I have lists A and B, I removed A[0], A[1], and B[last] and the algorithm with maps said that there is the diff in only 1 element, when in fact it's there 3 mutual exclusive elements.

The original code isn't a piece of art but it's fine tuned and it does its job. Given that we are not going to do any sort of performance tests using these test case scenarios I don't see any potential risk for this piece of code even though it's not perfect.

**Point**
Speaking about money and cost of maintenance, this PR is in review for almost 2 weeks and it blocks the team of 3 people. I think, that the cost of blockage is already way higher than any potential maintenance of, effectively, 8 lines of code, 4 of them are pretty much logging and reporting to CLI.